src/main/java/com/google/sps/servlets/ContestsServlet.java

```diff
@@ -62,30 +64,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
 
     Election election = Election.fromEntity(electionEntityOptional.get());
-    
-    List<String> contestJsonList =
+
+    List<JsonElement> contestJsonList =
         ImmutableSet.copyOf(election.getContests())
             .stream()
             .map(id -> KeyFactory.createKey(Contest.ENTITY_KIND, id.longValue()))
             .map(key -> ServletUtils.getFromDatastore(datastore, key))
-            .map(entity -> Contest.fromEntity(entity).toJsonString(datastore))
+            .map(
+                entity ->
+                    JsonParser.parseString(Contest.fromEntity(entity).toJsonString(datastore)))
             .collect(ImmutableList.toImmutableList());
 
-    List<String> referendumJsonList =
+    List<JsonElement> referendumJsonList =
         ImmutableSet.copyOf(election.getReferendums())
             .stream()
             .map(id -> KeyFactory.createKey(Referendum.ENTITY_KIND, id.longValue()))
             .map(key -> ServletUtils.getFromDatastore(datastore, key))
-            .map(entity -> Referendum.fromEntity(entity).toJsonString())
+            .map(entity -> JsonParser.parseString(Referendum.fromEntity(entity).toJsonString()))
             .collect(ImmutableList.toImmutableList());
 
     // The toJson function reserializes the already serialized JSON strings, so remove the extra
     // escaped characters.
     Gson gson = new Gson();
-    String contestJson =
-        gson.toJson(contestJsonList).replace("\\", "").replace("\"{", "{").replace("}\"", "}");
```

Comment:
I think having escaped quotes within the JSON object is still important in order to maintain cases like Edmund "Jerry" Brown on the site. I used replaceAll("\"","\\\\\"") when reading the Candidate name from the JSON object, but I'll try this as well.